### DISCUSSION TITLE
proxy websockets

### DISCUSSION THREAD

**Discussion by prabirshrestha** [trillium-rs/trillium#211] [Q&A]
I have create-react-app and have the proxy working for http, but for websockets it is failing.
https://github.com/prabirshrestha/objstor/tree/baace9364171f63a66f4826d65a092ba7216a147

```rust
pub fn spa() -> impl Handler {
    use trillium_rustls::RustlsConnector;
    use trillium_tokio::TcpConnector;
    type Proxy = trillium_proxy::Proxy<RustlsConnector<TcpConnector>>;
    Proxy::new("http://localhost:3000")
}
```

I tried registering a wildcard route with `Router::new().get("*", spa())` and the without using `spa()` but I'm not able to proxy websocket connections. I get the following error.

![image](https://user-images.githubusercontent.com/287744/173512322-33dbe653-926c-43d6-b955-8b83c73baf30.png)

Is it possible to proxy websocket connections?

**Comment by jbr** [comment]
It should be possible to proxy websocket connections. I don't understand why there's a router involved here, though. Why not just inline `spa()`?

Additionally, why are you using a RustlsConnector when you're not making any upstream tls connections? You can just use `Proxy<TcpConnector>`

**Reply by prabirshrestha** [reply]
Made the appropriate changes but still not able to connector to websockets.

https://github.com/prabirshrestha/objstor/commit/d015f961ad216ab2431c0a5c4f803d0150d699f8
